### ISSUE TITLE
when editing an existing object, Cancel button must be clicked twice

### ISSUE THREAD

**Issue by larkin** [nasa/openmct#1013]
When editing an existing object and trying to cancel, you must click the cancel button twice.

Currently appearing in VISTA, diagnosing with @akhenry

**Comment by larkin** [comment]
Seems that both save and cancel are both failing to work.

Boils down to this line of code:

**Comment by larkin** [comment]
Seems that both save and cancel are both failing to work.

Boils down to this line of code:

https://github.com/nasa/openmct/commit/b7765ff388a0eb23b3435a40c8498de374bcfa01

domainObject.getId() is not a valid persistence identifier anymore-- it may also contain a persistence space.  It needs to be parsed before attempting to fetch an object with it.

**Comment by larkin** [comment]
tldr; refresh tries to fetch from persistence using `persistenceCapability.getSpace()` and `domainObject.getId()`, when it should really be using the identifier service to parse the id and use the resulting id.

**Comment by larkin** [comment]
Instructions for verification in VISTA:

**Discarding changes**

1. edit existing object
2. make changes to object
3. press cancel

*Expected outcome: changes to object should be reverted, object should exit edit mode.*

**Saving edits**

1. edit existing object
2. press save

*Expected outcome: Object should save, VISTA should exit edit mode.*

**Comment by charlesh88** [comment]
Testathon 2016-06-16 using ciedevweb1.jpl.nasa.gov:8443/mcws/clients/vista/
Partially fixed:
* **Cancel must still be clicked twice**. Additionally, when clicking Cancel twice, my changes are "saved". I removed a telem elem from a stacked plot, and after exiting (even though I cancelled twice) my telem elem was gone. Refreshing VISTA restores the "removed" object.
* **Save appears to be working as expected**.
